pkg/webhook/webhook.go

```diff
@@ -36,6 +37,23 @@ type combinedDrainData struct {
 	InstanceID string
 }
 
+var instanceEventMap = make(map[string]*monitor.InterruptionEvent)
+var m sync.RWMutex
+
+type PostDecorator func(additionalInfo ec2metadata.NodeMetadata, event *monitor.InterruptionEvent, nthConfig config.Config)
+
+func SinglePost(postFn PostDecorator) PostDecorator {
```

Comment:
The name is a little misleading -- e.g. does the decorated function only ever post a single webhook?  How about `DedupeByInstanceId`?